src/app/admin/subject/[slug]/page.tsx

```diff
@@ -0,0 +1,185 @@
+"use client";
+import { getUser } from "@/components/hooks/users";
+import SubjectSidebar from "@/components/subjectHomepage/subject-sidebar";
+import Footer from "@/components/ui/footer";
+import Navbar from "@/components/ui/navbar";
+import { db } from "@/lib/firebase";
+import { Subject } from "@/types";
+import { User } from "@/types/user";
+import { doc, getDoc, setDoc } from "firebase/firestore";
+import { useEffect, useState } from "react";
+
+
+const apClasses = [
+  "AP 2-D Art and Design",
+  "AP 3-D Art and Design",
+  "AP Art History",
+  "AP Biology",
+  "AP Calculus AB",
+  "AP Calculus BC",
+  "AP Chemistry",
+  "AP Chinese",
+  "AP Comparative Government",
+  "AP Computer Science A",
+  "AP Computer Science Principles",
+  "AP Drawing",
+  "AP English Language",
+  "AP English Literature",
+  "AP Environmental Science",
+  "AP European History",
+  "AP French",
+  "AP German",
+  "AP Human Geography",
+  "AP Italian",
+  "AP Japanese",
+  "AP Latin",
+  "AP Macroeconomics",
+  "AP Microeconomics",
+  "AP Music Theory",
+  "AP Physics 1",
+  "AP Physics 2",
+  "AP Physics C: E&M",
+  "AP Physics C: Mechanics",
+  "AP Precalculus",
+  "AP Psychology",
+  "AP Research",
+  "AP Seminar",
+  "AP Spanish Language",
+  "AP Spanish Literature",
+  "AP Statistics",
+  "AP US History",
+  "AP United States Government",
+  "AP World History: Modern",
+].sort();
```

Comment:
Case in point, this should be in some central file so you don't need to copypaste it